backend/src/helpers/organization.ts

```diff
@@ -133,24 +138,76 @@ const createOrganization = async ({
 				description: name
 			});
 
-			organization = await new Organization({
-				name,
-				customerId: customer.id
-			}).save();
-		} else {
-			organization = await new Organization({
-				name
-			}).save();
-		}
+            organization = await new Organization({
+                name,
+                customerId: customer.id
+            }).save();
+        } else {
+            organization = await new Organization({
+                name
+            }).save();
+        }
 
-		await initSubscriptionOrg({ organizationId: organization._id });
-	} catch (err) {
-		Sentry.setUser({ email });
-		Sentry.captureException(err);
-		throw new Error(`Failed to create organization [err=${err}]`);
-	}
+        await initSubscriptionOrg({ organizationId: organization._id });
+    } catch (err) {
+        Sentry.setUser({ email });
+        Sentry.captureException(err);
+        throw new Error(`Failed to create organization [err=${err}]`);
+    }
+
+    return organization;
+};
+
+const deleteOrganization = async ({
+    email,
+    orgId
+}: {
+    email: string;
+    orgId: string;
+}) => {
+    let organization;
+    try {
+        // delete the organization itself
+        const organization = await Organization.findByIdAndDelete(orgId);
 
-	return organization;
+        // delete all the membersOrg
+        await MembershipOrg.deleteMany({
+            organization: orgId
+        });
+
+        // delete all the workspaces
+        // first get all the workspaces ids
+        const workspaceIds = (
+            await Workspace.find({
+                organization: orgId
+            })
+        ).map((m) => m.id);
+
+        // delete all the workspaces one by one
+        for (const id of workspaceIds) {
+            await deleteWorkspace({ id });
+        }
+
+        // delete the stripe customer
+        const stripe = new Stripe(await getStripeSecretKey(), {
+            apiVersion: '2022-08-01'
+        });
+
+        if (await getStripeSecretKey()) {
+            const customer = await stripe.customers.list({
+                email: email,
```

Comment:
So the email of the user on the request isn't actually the Stripe customer you want to delete; the Stripe customer you want to delete is the one with id `customerId` on the organization model.